File: backend-go/projects/gateway/internal/gatewaysession/registry.go
package gatewaysession

// ListGatewaySessionIdentityResolvers mirrors listGatewaySessionIdentityResolvers.
func ListGatewaySessionIdentityResolvers() []Resolver {
	return DefaultGatewaySessionIdentityResolvers
}

// CollectGatewaySessionIdentityCandidates mirrors
// collectGatewaySessionIdentityCandidates: resolvers == nil selects the
// default resolver set; candidate order is resolver order, then collection
// order (the Node flatMap order).
func CollectGatewaySessionIdentityCandidates(context ResolverContext, resolvers []Resolver) []RawCandidate {
	if resolvers == nil {
		resolvers = DefaultGatewaySessionIdentityResolvers
	}
	var candidates []RawCandidate
	for _, resolver := range resolvers {
		candidates = append(candidates, resolver.Collect(context)...)
	}
	return candidates
}

File: migration-backup-1/node/final-archive/backend/src/modules/gateway/session-identity/registry.ts
import { defaultGatewaySessionIdentityResolvers } from './resolvers.js'
import type {
  GatewaySessionIdentityRawCandidate,
  GatewaySessionIdentityResolver,
  GatewaySessionIdentityResolverContext
} from './types.js'

export function listGatewaySessionIdentityResolvers(): readonly GatewaySessionIdentityResolver[] {
  return defaultGatewaySessionIdentityResolvers
}

export function collectGatewaySessionIdentityCandidates(
  context: GatewaySessionIdentityResolverContext,
  resolvers: readonly GatewaySessionIdentityResolver[] = defaultGatewaySessionIdentityResolvers
): GatewaySessionIdentityRawCandidate[] {
  return resolvers.flatMap((resolver) => resolver.collect(context))
}

